src/Blockcore.Explorer/ClientApp/src/app/explorer/contractaddress/contractaddress.component.ts

```diff
@@ -0,0 +1,156 @@
+import { Component, HostBinding, OnInit, OnDestroy, HostListener } from '@angular/core';
+import { ActivatedRoute, Route, Router } from '@angular/router';
+import { ApiComponent } from 'src/app/api/api.component';
+import { ApiService, HttpError } from 'src/app/services/api.service';
+import { SetupService } from 'src/app/services/setup.service';
+import { ScrollEvent } from 'src/app/shared/scroll.directive';
+
+@Component({
+   selector: 'app-contractaddress-component',
+   templateUrl: './contractaddress.component.html'
+})
+export class ContractAddressComponent implements OnInit, OnDestroy {
+   @HostBinding('class.content-centered-top') hostClass = true;
+
+   info: any;
+   node: any;
+   blockchain: any;
+   network: any;
+   configuration: any;
+   consensus: any;
+   peers: any;
+   blocks: any;
+   transactions: any;
+   transaction: any;
+
+   timerInfo: any;
+   timerBlocks: any;
+   timerTransactions: any;
+   address: any;
+   balance: any;
+   detailsVisible = false;
+   lastBlockHeight: number;
+   subscription: any;
+   limit = 10;
+   loading = false;
+   count = 0;
+   total: any;
+   link: string;
+   error: any;
+   errorTransactions: any;
+
+   constructor(
+      private api: ApiService,
+      private router: Router,
+      public setup: SetupService,
+      private activatedRoute: ActivatedRoute) {
+
+      this.activatedRoute.paramMap.subscribe(async params => {
+         const id: any = params.get('address');
+         console.log('Address:', id);
```

Comment:
We should introduce a logger :-)

Reply:
you mean a different one to `console.log()`